// @pulumi/aws/ecs/taskSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Import
 *
 * ECS Task Sets can be imported via the `task_set_id`, `service`, and `cluster` separated by commas (`,`) e.g.
 *
 * ```sh
 *  $ pulumi import aws:ecs/taskSet:TaskSet example ecs-svc/7177320696926227436,arn:aws:ecs:us-west-2:123456789101:service/example/example-1234567890,arn:aws:ecs:us-west-2:123456789101:cluster/example
 * ```
 */
class TaskSet extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["arn"] = state ? state.arn : undefined;
            inputs["capacityProviderStrategies"] = state ? state.capacityProviderStrategies : undefined;
            inputs["cluster"] = state ? state.cluster : undefined;
            inputs["externalId"] = state ? state.externalId : undefined;
            inputs["forceDelete"] = state ? state.forceDelete : undefined;
            inputs["launchType"] = state ? state.launchType : undefined;
            inputs["loadBalancers"] = state ? state.loadBalancers : undefined;
            inputs["networkConfiguration"] = state ? state.networkConfiguration : undefined;
            inputs["platformVersion"] = state ? state.platformVersion : undefined;
            inputs["scale"] = state ? state.scale : undefined;
            inputs["service"] = state ? state.service : undefined;
            inputs["serviceRegistries"] = state ? state.serviceRegistries : undefined;
            inputs["stabilityStatus"] = state ? state.stabilityStatus : undefined;
            inputs["status"] = state ? state.status : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["tagsAll"] = state ? state.tagsAll : undefined;
            inputs["taskDefinition"] = state ? state.taskDefinition : undefined;
            inputs["taskSetId"] = state ? state.taskSetId : undefined;
            inputs["waitUntilStable"] = state ? state.waitUntilStable : undefined;
            inputs["waitUntilStableTimeout"] = state ? state.waitUntilStableTimeout : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.cluster === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cluster'");
            }
            if ((!args || args.service === undefined) && !opts.urn) {
                throw new Error("Missing required property 'service'");
            }
            if ((!args || args.taskDefinition === undefined) && !opts.urn) {
                throw new Error("Missing required property 'taskDefinition'");
            }
            inputs["capacityProviderStrategies"] = args ? args.capacityProviderStrategies : undefined;
            inputs["cluster"] = args ? args.cluster : undefined;
            inputs["externalId"] = args ? args.externalId : undefined;
            inputs["forceDelete"] = args ? args.forceDelete : undefined;
            inputs["launchType"] = args ? args.launchType : undefined;
            inputs["loadBalancers"] = args ? args.loadBalancers : undefined;
            inputs["networkConfiguration"] = args ? args.networkConfiguration : undefined;
            inputs["platformVersion"] = args ? args.platformVersion : undefined;
            inputs["scale"] = args ? args.scale : undefined;
            inputs["service"] = args ? args.service : undefined;
            inputs["serviceRegistries"] = args ? args.serviceRegistries : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tagsAll"] = args ? args.tagsAll : undefined;
            inputs["taskDefinition"] = args ? args.taskDefinition : undefined;
            inputs["waitUntilStable"] = args ? args.waitUntilStable : undefined;
            inputs["waitUntilStableTimeout"] = args ? args.waitUntilStableTimeout : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["stabilityStatus"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["taskSetId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(TaskSet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TaskSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TaskSet(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TaskSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TaskSet.__pulumiType;
    }
}
exports.TaskSet = TaskSet;
/** @internal */
TaskSet.__pulumiType = 'aws:ecs/taskSet:TaskSet';
