// @pulumi/aws/connect/hoursOfOperation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HoursOfOperation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an Amazon Connect Hours of Operation resource. For more information see
 * [Amazon Connect: Getting Started](https://docs.aws.amazon.com/connect/latest/adminguide/amazon-connect-get-started.html)
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.connect.HoursOfOperation("test", {
 *     configs: [
 *         {
 *             day: "MONDAY",
 *             endTime: {
 *                 hours: 23,
 *                 minutes: 8,
 *             },
 *             startTime: {
 *                 hours: 8,
 *                 minutes: 0,
 *             },
 *         },
 *         {
 *             day: "TUESDAY",
 *             endTime: {
 *                 hours: 21,
 *                 minutes: 0,
 *             },
 *             startTime: {
 *                 hours: 9,
 *                 minutes: 0,
 *             },
 *         },
 *     ],
 *     description: "Monday office hours",
 *     instanceId: "aaaaaaaa-bbbb-cccc-dddd-111111111111",
 *     tags: {
 *         Name: "Example Hours of Operation",
 *     },
 *     timeZone: "EST",
 * });
 * ```
 *
 * ## Import
 *
 * Amazon Connect Hours of Operations can be imported using the `instance_id` and `hours_of_operation_id` separated by a colon (`:`), e.g.,
 *
 * ```sh
 *  $ pulumi import aws:connect/hoursOfOperation:HoursOfOperation example f1288a1f-6193-445a-b47e-af739b2:c1d4e5f6-1b3c-1b3c-1b3c-c1d4e5f6c1d4e5
 * ```
 */
class HoursOfOperation extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["configs"] = state ? state.configs : undefined;
            inputs["description"] = state ? state.description : undefined;
            inputs["hoursOfOperationArn"] = state ? state.hoursOfOperationArn : undefined;
            inputs["hoursOfOperationId"] = state ? state.hoursOfOperationId : undefined;
            inputs["instanceId"] = state ? state.instanceId : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["tagsAll"] = state ? state.tagsAll : undefined;
            inputs["timeZone"] = state ? state.timeZone : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.configs === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configs'");
            }
            if ((!args || args.instanceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceId'");
            }
            if ((!args || args.timeZone === undefined) && !opts.urn) {
                throw new Error("Missing required property 'timeZone'");
            }
            inputs["configs"] = args ? args.configs : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["instanceId"] = args ? args.instanceId : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tagsAll"] = args ? args.tagsAll : undefined;
            inputs["timeZone"] = args ? args.timeZone : undefined;
            inputs["hoursOfOperationArn"] = undefined /*out*/;
            inputs["hoursOfOperationId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(HoursOfOperation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing HoursOfOperation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HoursOfOperation(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HoursOfOperation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HoursOfOperation.__pulumiType;
    }
}
exports.HoursOfOperation = HoursOfOperation;
/** @internal */
HoursOfOperation.__pulumiType = 'aws:connect/hoursOfOperation:HoursOfOperation';
