// @pulumi/aws/fsx/openZfsVolume.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenZfsVolume = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Amazon FSx for OpenZFS volume.
 * See the [FSx OpenZFS User Guide](https://docs.aws.amazon.com/fsx/latest/OpenZFSGuide/what-is-fsx.html) for more information.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.fsx.OpenZfsVolume("test", {parentVolumeId: aws_fsx_openzfs_file_system.test.root_volume_id});
 * ```
 *
 * ## Import
 *
 * FSx Volumes can be imported using the `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:fsx/openZfsVolume:OpenZfsVolume example fsvol-543ab12b1ca672f33
 * ```
 */
class OpenZfsVolume extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["arn"] = state ? state.arn : undefined;
            inputs["copyTagsToSnapshots"] = state ? state.copyTagsToSnapshots : undefined;
            inputs["dataCompressionType"] = state ? state.dataCompressionType : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["nfsExports"] = state ? state.nfsExports : undefined;
            inputs["originSnapshot"] = state ? state.originSnapshot : undefined;
            inputs["parentVolumeId"] = state ? state.parentVolumeId : undefined;
            inputs["readOnly"] = state ? state.readOnly : undefined;
            inputs["storageCapacityQuotaGib"] = state ? state.storageCapacityQuotaGib : undefined;
            inputs["storageCapacityReservationGib"] = state ? state.storageCapacityReservationGib : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["tagsAll"] = state ? state.tagsAll : undefined;
            inputs["userAndGroupQuotas"] = state ? state.userAndGroupQuotas : undefined;
            inputs["volumeType"] = state ? state.volumeType : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.parentVolumeId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'parentVolumeId'");
            }
            inputs["copyTagsToSnapshots"] = args ? args.copyTagsToSnapshots : undefined;
            inputs["dataCompressionType"] = args ? args.dataCompressionType : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["nfsExports"] = args ? args.nfsExports : undefined;
            inputs["originSnapshot"] = args ? args.originSnapshot : undefined;
            inputs["parentVolumeId"] = args ? args.parentVolumeId : undefined;
            inputs["readOnly"] = args ? args.readOnly : undefined;
            inputs["storageCapacityQuotaGib"] = args ? args.storageCapacityQuotaGib : undefined;
            inputs["storageCapacityReservationGib"] = args ? args.storageCapacityReservationGib : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tagsAll"] = args ? args.tagsAll : undefined;
            inputs["userAndGroupQuotas"] = args ? args.userAndGroupQuotas : undefined;
            inputs["volumeType"] = args ? args.volumeType : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(OpenZfsVolume.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing OpenZfsVolume resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new OpenZfsVolume(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OpenZfsVolume.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OpenZfsVolume.__pulumiType;
    }
}
exports.OpenZfsVolume = OpenZfsVolume;
/** @internal */
OpenZfsVolume.__pulumiType = 'aws:fsx/openZfsVolume:OpenZfsVolume';
