// @pulumi/aws/ec2/vpcIpamPool.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcIpamPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an IP address pool resource for IPAM.
 *
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const current = aws.getRegion({});
 * const exampleVpcIpam = new aws.ec2.VpcIpam("exampleVpcIpam", {operatingRegions: [{
 *     regionName: current.then(current => current.name),
 * }]});
 * const exampleVpcIpamPool = new aws.ec2.VpcIpamPool("exampleVpcIpamPool", {
 *     addressFamily: "ipv4",
 *     ipamScopeId: exampleVpcIpam.privateDefaultScopeId,
 *     locale: current.then(current => current.name),
 * });
 * ```
 *
 * Nested Pools:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const current = aws.getRegion({});
 * const example = new aws.ec2.VpcIpam("example", {operatingRegions: [{
 *     regionName: current.then(current => current.name),
 * }]});
 * const parent = new aws.ec2.VpcIpamPool("parent", {
 *     addressFamily: "ipv4",
 *     ipamScopeId: example.privateDefaultScopeId,
 * });
 * const parentTest = new aws.ec2.VpcIpamPoolCidr("parentTest", {
 *     ipamPoolId: parent.id,
 *     cidr: "172.2.0.0/16",
 * });
 * const child = new aws.ec2.VpcIpamPool("child", {
 *     addressFamily: "ipv4",
 *     ipamScopeId: example.privateDefaultScopeId,
 *     locale: current.then(current => current.name),
 *     sourceIpamPoolId: parent.id,
 * });
 * const childTest = new aws.ec2.VpcIpamPoolCidr("childTest", {
 *     ipamPoolId: child.id,
 *     cidr: "172.2.0.0/24",
 * });
 * ```
 *
 * ## Import
 *
 * IPAMs can be imported using the `ipam pool id`, e.g.
 *
 * ```sh
 *  $ pulumi import aws:ec2/vpcIpamPool:VpcIpamPool example ipam-pool-0958f95207d978e1e
 * ```
 */
class VpcIpamPool extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["addressFamily"] = state ? state.addressFamily : undefined;
            inputs["allocationDefaultNetmaskLength"] = state ? state.allocationDefaultNetmaskLength : undefined;
            inputs["allocationMaxNetmaskLength"] = state ? state.allocationMaxNetmaskLength : undefined;
            inputs["allocationMinNetmaskLength"] = state ? state.allocationMinNetmaskLength : undefined;
            inputs["allocationResourceTags"] = state ? state.allocationResourceTags : undefined;
            inputs["arn"] = state ? state.arn : undefined;
            inputs["autoImport"] = state ? state.autoImport : undefined;
            inputs["awsService"] = state ? state.awsService : undefined;
            inputs["description"] = state ? state.description : undefined;
            inputs["ipamScopeId"] = state ? state.ipamScopeId : undefined;
            inputs["ipamScopeType"] = state ? state.ipamScopeType : undefined;
            inputs["locale"] = state ? state.locale : undefined;
            inputs["poolDepth"] = state ? state.poolDepth : undefined;
            inputs["publiclyAdvertisable"] = state ? state.publiclyAdvertisable : undefined;
            inputs["sourceIpamPoolId"] = state ? state.sourceIpamPoolId : undefined;
            inputs["state"] = state ? state.state : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["tagsAll"] = state ? state.tagsAll : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.addressFamily === undefined) && !opts.urn) {
                throw new Error("Missing required property 'addressFamily'");
            }
            if ((!args || args.ipamScopeId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ipamScopeId'");
            }
            inputs["addressFamily"] = args ? args.addressFamily : undefined;
            inputs["allocationDefaultNetmaskLength"] = args ? args.allocationDefaultNetmaskLength : undefined;
            inputs["allocationMaxNetmaskLength"] = args ? args.allocationMaxNetmaskLength : undefined;
            inputs["allocationMinNetmaskLength"] = args ? args.allocationMinNetmaskLength : undefined;
            inputs["allocationResourceTags"] = args ? args.allocationResourceTags : undefined;
            inputs["autoImport"] = args ? args.autoImport : undefined;
            inputs["awsService"] = args ? args.awsService : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["ipamScopeId"] = args ? args.ipamScopeId : undefined;
            inputs["locale"] = args ? args.locale : undefined;
            inputs["publiclyAdvertisable"] = args ? args.publiclyAdvertisable : undefined;
            inputs["sourceIpamPoolId"] = args ? args.sourceIpamPoolId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["ipamScopeType"] = undefined /*out*/;
            inputs["poolDepth"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["tagsAll"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(VpcIpamPool.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VpcIpamPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcIpamPool(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VpcIpamPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcIpamPool.__pulumiType;
    }
}
exports.VpcIpamPool = VpcIpamPool;
/** @internal */
VpcIpamPool.__pulumiType = 'aws:ec2/vpcIpamPool:VpcIpamPool';
