// @pulumi/aws/codecommit/approvalRuleTemplate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApprovalRuleTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a CodeCommit Approval Rule Template Resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.codecommit.ApprovalRuleTemplate("example", {
 *     content: `{
 *     "Version": "2018-11-08",
 *     "DestinationReferences": ["refs/heads/master"],
 *     "Statements": [{
 *         "Type": "Approvers",
 *         "NumberOfApprovalsNeeded": 2,
 *         "ApprovalPoolMembers": ["arn:aws:sts::123456789012:assumed-role/CodeCommitReview/*"]
 *     }]
 * }
 * `,
 *     description: "This is an example approval rule template",
 * });
 * ```
 *
 * ## Import
 *
 * CodeCommit approval rule templates can be imported using the `name`, e.g.
 *
 * ```sh
 *  $ pulumi import aws:codecommit/approvalRuleTemplate:ApprovalRuleTemplate imported ExistingApprovalRuleTemplateName
 * ```
 */
class ApprovalRuleTemplate extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["approvalRuleTemplateId"] = state ? state.approvalRuleTemplateId : undefined;
            inputs["content"] = state ? state.content : undefined;
            inputs["creationDate"] = state ? state.creationDate : undefined;
            inputs["description"] = state ? state.description : undefined;
            inputs["lastModifiedDate"] = state ? state.lastModifiedDate : undefined;
            inputs["lastModifiedUser"] = state ? state.lastModifiedUser : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["ruleContentSha256"] = state ? state.ruleContentSha256 : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.content === undefined) && !opts.urn) {
                throw new Error("Missing required property 'content'");
            }
            inputs["content"] = args ? args.content : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["approvalRuleTemplateId"] = undefined /*out*/;
            inputs["creationDate"] = undefined /*out*/;
            inputs["lastModifiedDate"] = undefined /*out*/;
            inputs["lastModifiedUser"] = undefined /*out*/;
            inputs["ruleContentSha256"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ApprovalRuleTemplate.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ApprovalRuleTemplate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ApprovalRuleTemplate(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApprovalRuleTemplate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApprovalRuleTemplate.__pulumiType;
    }
}
exports.ApprovalRuleTemplate = ApprovalRuleTemplate;
/** @internal */
ApprovalRuleTemplate.__pulumiType = 'aws:codecommit/approvalRuleTemplate:ApprovalRuleTemplate';
