// @pulumi/aws/emr/studioSessionMapping.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioSessionMapping = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an Elastic MapReduce Studio Session Mapping.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.emr.StudioSessionMapping("example", {
 *     studioId: aws_emr_studio.example.id,
 *     identityType: "USER",
 *     identityId: "example",
 *     sessionPolicyArn: aws_iam_policy.example.arn,
 * });
 * ```
 *
 * ## Import
 *
 * EMR studio session mappings can be imported using the `id`, e.g., `studio-id:identity-type:identity-id`
 *
 * ```sh
 *  $ pulumi import aws:emr/studioSessionMapping:StudioSessionMapping example es-xxxxx:USER:xxxxx-xxx-xxx
 * ```
 */
class StudioSessionMapping extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["identityId"] = state ? state.identityId : undefined;
            inputs["identityName"] = state ? state.identityName : undefined;
            inputs["identityType"] = state ? state.identityType : undefined;
            inputs["sessionPolicyArn"] = state ? state.sessionPolicyArn : undefined;
            inputs["studioId"] = state ? state.studioId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.identityType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identityType'");
            }
            if ((!args || args.sessionPolicyArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sessionPolicyArn'");
            }
            if ((!args || args.studioId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'studioId'");
            }
            inputs["identityId"] = args ? args.identityId : undefined;
            inputs["identityName"] = args ? args.identityName : undefined;
            inputs["identityType"] = args ? args.identityType : undefined;
            inputs["sessionPolicyArn"] = args ? args.sessionPolicyArn : undefined;
            inputs["studioId"] = args ? args.studioId : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(StudioSessionMapping.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StudioSessionMapping resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StudioSessionMapping(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StudioSessionMapping.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StudioSessionMapping.__pulumiType;
    }
}
exports.StudioSessionMapping = StudioSessionMapping;
/** @internal */
StudioSessionMapping.__pulumiType = 'aws:emr/studioSessionMapping:StudioSessionMapping';
