// @pulumi/aws/emr/studio.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Studio = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an Elastic MapReduce Studio.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.emr.Studio("example", {
 *     authMode: "SSO",
 *     defaultS3Location: `s3://${aws_s3_bucket.test.bucket}/test`,
 *     engineSecurityGroupId: aws_security_group.test.id,
 *     serviceRole: aws_iam_role.test.arn,
 *     subnetIds: [aws_subnet.test.id],
 *     userRole: aws_iam_role.test.arn,
 *     vpcId: aws_vpc.test.id,
 *     workspaceSecurityGroupId: aws_security_group.test.id,
 * });
 * ```
 *
 * ## Import
 *
 * EMR studios can be imported using the `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:emr/studio:Studio studio es-123456ABCDEF
 * ```
 */
class Studio extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["arn"] = state ? state.arn : undefined;
            inputs["authMode"] = state ? state.authMode : undefined;
            inputs["defaultS3Location"] = state ? state.defaultS3Location : undefined;
            inputs["description"] = state ? state.description : undefined;
            inputs["engineSecurityGroupId"] = state ? state.engineSecurityGroupId : undefined;
            inputs["idpAuthUrl"] = state ? state.idpAuthUrl : undefined;
            inputs["idpRelayStateParameterName"] = state ? state.idpRelayStateParameterName : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["serviceRole"] = state ? state.serviceRole : undefined;
            inputs["subnetIds"] = state ? state.subnetIds : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["tagsAll"] = state ? state.tagsAll : undefined;
            inputs["url"] = state ? state.url : undefined;
            inputs["userRole"] = state ? state.userRole : undefined;
            inputs["vpcId"] = state ? state.vpcId : undefined;
            inputs["workspaceSecurityGroupId"] = state ? state.workspaceSecurityGroupId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.authMode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authMode'");
            }
            if ((!args || args.defaultS3Location === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultS3Location'");
            }
            if ((!args || args.engineSecurityGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'engineSecurityGroupId'");
            }
            if ((!args || args.serviceRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceRole'");
            }
            if ((!args || args.subnetIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetIds'");
            }
            if ((!args || args.vpcId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcId'");
            }
            if ((!args || args.workspaceSecurityGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceSecurityGroupId'");
            }
            inputs["authMode"] = args ? args.authMode : undefined;
            inputs["defaultS3Location"] = args ? args.defaultS3Location : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["engineSecurityGroupId"] = args ? args.engineSecurityGroupId : undefined;
            inputs["idpAuthUrl"] = args ? args.idpAuthUrl : undefined;
            inputs["idpRelayStateParameterName"] = args ? args.idpRelayStateParameterName : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["serviceRole"] = args ? args.serviceRole : undefined;
            inputs["subnetIds"] = args ? args.subnetIds : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tagsAll"] = args ? args.tagsAll : undefined;
            inputs["userRole"] = args ? args.userRole : undefined;
            inputs["vpcId"] = args ? args.vpcId : undefined;
            inputs["workspaceSecurityGroupId"] = args ? args.workspaceSecurityGroupId : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["url"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Studio.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Studio resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Studio(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Studio.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Studio.__pulumiType;
    }
}
exports.Studio = Studio;
/** @internal */
Studio.__pulumiType = 'aws:emr/studio:Studio';
