// @pulumi/aws/account/alternativeContact.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlternativeContact = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages the specified alternate contact attached to an AWS Account.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const operations = new aws.account.AlternativeContact("operations", {
 *     alternateContactType: "OPERATIONS",
 *     emailAddress: "test@example.com",
 *     phoneNumber: "+1234567890",
 *     title: "Example",
 * });
 * ```
 *
 * ## Import
 *
 * The current Alternate Contact can be imported using the `alternate_contact_type`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:account/alternativeContact:AlternativeContact operations OPERATIONS
 * ```
 */
class AlternativeContact extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["accountId"] = state ? state.accountId : undefined;
            inputs["alternateContactType"] = state ? state.alternateContactType : undefined;
            inputs["emailAddress"] = state ? state.emailAddress : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["phoneNumber"] = state ? state.phoneNumber : undefined;
            inputs["title"] = state ? state.title : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.alternateContactType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alternateContactType'");
            }
            if ((!args || args.emailAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'emailAddress'");
            }
            if ((!args || args.phoneNumber === undefined) && !opts.urn) {
                throw new Error("Missing required property 'phoneNumber'");
            }
            if ((!args || args.title === undefined) && !opts.urn) {
                throw new Error("Missing required property 'title'");
            }
            inputs["accountId"] = args ? args.accountId : undefined;
            inputs["alternateContactType"] = args ? args.alternateContactType : undefined;
            inputs["emailAddress"] = args ? args.emailAddress : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["phoneNumber"] = args ? args.phoneNumber : undefined;
            inputs["title"] = args ? args.title : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(AlternativeContact.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AlternativeContact resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AlternativeContact(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AlternativeContact.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AlternativeContact.__pulumiType;
    }
}
exports.AlternativeContact = AlternativeContact;
/** @internal */
AlternativeContact.__pulumiType = 'aws:account/alternativeContact:AlternativeContact';
